includes/abstracts/abstract-wc-gateway-ppec.php

```diff
@@ -209,7 +209,7 @@ private function get_certificate_info( $cert_string ) {
 				}
 
 				$expires = sprintf( $expires, date_i18n( get_option( 'date_format' ), $valid_until ) );
-				$out = sprintf( __( 'Certificate belongs to API username %1$s; %2$s', 'woocommerce-gateway-paypal-express-checkout' ), $certinfo['subject']['CN'], $expires );
+				$out = sprintf( __( 'Certificate belongs to API username %1$s; %2$s.', 'woocommerce-gateway-paypal-express-checkout' ), $certinfo['subject']['CN'], $expires );
 			} else {
 				$out = __( 'The certificate on file is not valid.', 'woocommerce-gateway-paypal-express-checkout' );
```

Comment:
Would you know what settings would bring this text to be displayed? I tried a few different steps, but I could not get this message.
1. After a certificate is successfully set, I removed it from the PayPal Account settings.
2. I requested for a new certificate.
3. Now, I saved the WC PPEC settings page. But I cannot see this message.

Reply:
Yeah, that's a good question. I thought it would be shown if there was a file but it was invalid. When I uploaded an invalid file it didn't show anything though. 

eg: 
<img width="533" alt="Screen Shot 2020-03-31 at 10 46 45 am" src="https://user-images.githubusercontent.com/8490476/77975264-f0003980-733c-11ea-8104-a84c49d849e0.png">

This is because when there's an invalid file, it doesn't pass the original `openssl_x509_read()` check. 

Looking at the code, this message would be shown if the cert can be read by `openssl_x509_read()` but cannot be parsed by `openssl_x509_parse()`, I don't know what circumstance would lead to that though.

For that reason, I've rejigged this part of the function in 42f6a28. 

----

Regarding your attempts to get this message. This notice is only displayed if the certificate is not a certificate file or it couldn't be parsed. It doesn't show if it's invalid as a credential on the account.

Reply:
Thanks for the clarification James.